### PR TITLE
Update the contributors documentation 

### PR DESCRIPTION
**Pull request by wallrj**
I corrected / removed some broken commands and some out of date advice in the documentation for contributors.

### CODE REVIEW

File: docs/development/submitting-patches.rst [hunk 1]
```diff
@@ -8,8 +8,6 @@ Submitting patches
   fork is also nice because that will you can work on more than one PR at a
   time.
 * Patches should be small to facilitate easier review.
-* New features and significant bug fixes should be documented in the
-  :doc:`/changelog`.
```
Comment (wallrj) [inline docs/development/submitting-patches.rst]: The changelog seems to be empty, so I don't see that this rule is being followed.
And I didn't see a changelog in https://pypi.python.org/pypi/mimic

File: docs/development/submitting-patches.rst [hunk 2]
```diff
@@ -27,28 +25,38 @@ You can check if your code meets our automated requirements by running
 
 `Write comments as complete sentences.`_
 
-Every code file must start with the boilerplate notice of the Apache License.
```
Comment (wallrj) [inline docs/development/submitting-patches.rst]: This doesn't seem to be followed by the existing modules and test modules.

File: docs/development/submitting-patches.rst [hunk 3]
```diff
@@ -27,28 +25,38 @@ You can check if your code meets our automated requirements by running
...
-Additionally, every Python code file must contain
+Every Python code file must contain:
 
 .. code-block:: python
 
-    from __future__ import absolute_import, division, print_function
+    from __future__ import absolute_import, division
```
Comment (wallrj) [inline docs/development/submitting-patches.rst]: existing modules don't import `print_function`

File: docs/development/submitting-patches.rst [hunk 4]
```diff
@@ -27,28 +25,38 @@ You can check if your code meets our automated requirements by running
...
+Then run:
+
+.. code-block:: console
+
+    $ coverage erase \
+      && detox \
+      && coverage combine \
+      && coverage html
+
+And open ``htmlcov/index.html`` in your web browser.
```
Comment (wallrj) [inline docs/development/submitting-patches.rst]: I took this from @glyph 's comment in https://github.com/rackerlabs/mimic/pull/375#issuecomment-155905468

File: docs/development/submitting-patches.rst [hunk 5]
```diff
@@ -67,6 +75,10 @@ run our ``tox`` job to lint docs.  We also provide a spell-check job for docs:
       docs-spellcheck: commands succeeded
       congratulations :)
 
+    $ tox -e docs-linkcheck
+      docs-linkcheck: commands succeeded
+      congratulations :)
```
Comment (wallrj) [inline docs/development/submitting-patches.rst]: Although, for some reason, this doesn't show up when I run:

```
$ tox --listenvs
py27
py34
py35
pypy
docs
lint
docs-spellcheck

```

File: requirements/toolchain.txt [hunk 6]
```diff
@@ -1,5 +1,6 @@
 # Tools for installing and setting up the environment.
 
+coverage==4.2
```
Comment (wallrj) [inline requirements/toolchain.txt]: This is already included in `production.txt` but is now needed outside of of the tox virtualenv.

File: tox.ini [hunk 7]
```diff
@@ -6,7 +6,7 @@ deps =
     --requirement={toxinidir}/requirements/production.txt
 passenv = PIP_WHEEL_DIR PIP_FIND_LINKS PIP_NO_INDEX
 commands =
-    coverage run -p {envbindir}/trial --rterrors {posargs:mimic}
+    coverage run --source mimic --branch --parallel-mode {envbindir}/trial --rterrors {posargs:mimic}
```
Comment (wallrj) [inline tox.ini]: This limits the coverage report to only the mimic modules, included branch coverage and significantly speeds up the tests.

File: docs/development/submitting-patches.rst (hunk 3 above)
Comment (glyph) [inline docs/development/submitting-patches.rst]: they probably _should_ though…

File: docs/development/submitting-patches.rst (hunk 4 above)
Comment (glyph) [inline docs/development/submitting-patches.rst]: Thanks!

File: tox.ini (hunk 7 above)
Comment (glyph) [inline tox.ini]: really this is a separate change, but I don't want to give you a hard time about it :)

**Pull request merged**